Documentation/content/docs/concepts_widgets.md

```diff
@@ -0,0 +1,123 @@
+title: Widgets
+---
+
+Widgets allow users to interact with scene objects and drive functionality
+through clicking, dragging, and touch.
+
+VTK.js currently has two flavors of widgets: the first, located in
+`Sources/Interaction/Widgets`, is the VTK C++ style widgets that were
+semi-ported over. The second, located in `Sources/Widgets`, is the new-style
+VTK.js widgets that were architected from the ground-up to address shortcomings
+with the original widget architecture.
+
+The rest of this document will be only concerned with the new-style widgets, as
+the old-style widgets are deprecated (but won't be removed in the near-term).
+
+## Widget Architecture Overview
+
+The widget architecture follows the Model-View-Controller (MVC) pattern. In
+mapping the MVC pattern to VTK.js widgets, there are three primary components
+involved:
+- `vtkWidgetState` is the model and the central source of truth for a widget.
+  When a widget is available across different renderers, they all refer to the
+  same widget state object for updates.
+- `vtkWidgetRepresentation` is the base class for the view, which listens to a
+  widget state object and generates the appropriate VTK.js scene objects that
+  represent the state.
+- `vtkAbstractWidget` is the base class that contains widget behavior,
+  interaction code, and a reference to the widget state. This object is closest
+  to the controller.
+
+In addition to the three components above, there are widget manipulators, a
+widget manager, and an abstract widget factory. In brief, manipulators take
+events (e.g. mouse events) and translate them into some meaningful 3D
+coordinate. The widget manager is a per-renderer manager that manages the
+lifecycle of widgets attached to one specific renderer. The abstract widget
+factory is the primary object that creates the widget state, constructs
+subclasses of `vtkAbstractWidget` to be attached to a widget manager, and
+constructs the associated representations with the view/renderer.
+
+*discuss general flow, maybe with a diagram*
```

Comment:
Yes, would <3 a diagram here :-)

Reply:
For diagram, I like to use http://interactive.blockdiag.com/